packages/apm/src/integrations/tracing.ts

```diff
@@ -336,11 +343,171 @@ export class Tracing implements Integration {
+            addPerformanceNavigationTiming(transactionSpan, entry, 'loadEvent');
+            addPerformanceNavigationTiming(transactionSpan, entry, 'connect');
+            addPerformanceNavigationTiming(transactionSpan, entry, 'domainLookup');
+            addRequest(transactionSpan, entry);
+            break;
+          case 'mark':
+          case 'paint':
+          case 'measure':
+            const mark = transactionSpan.child({
+              description: `${entry.entryType} ${entry.name}`,
+              op: 'mark',
+            });
+            mark.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+            mark.timestamp = mark.startTimestamp + duration;
+            if (tracingInitMarkStartTime === undefined && entry.name === 'sentry-tracing-init') {
+              tracingInitMarkStartTime = mark.startTimestamp;
+            }
+            addSpan(mark);
+            break;
+          case 'resource':
+            const resourceName = entry.name.replace(window.location.origin, '');
+            if (entry.initiatorType === 'xmlhttprequest' || entry.initiatorType === 'fetch') {
+              // We need to update existing spans with new timing info
+              // tslint:disable-next-line: no-non-null-assertion
+              transactionSpan.spanRecorder!.finishedSpans.map((finishedSpan: SpanClass) => {
+                if (finishedSpan.description && finishedSpan.description.indexOf(resourceName) !== -1) {
+                  finishedSpan.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+                  finishedSpan.timestamp = finishedSpan.startTimestamp + duration;
+                }
+              });
+            } else {
+              const resource = transactionSpan.child({
+                description: `${entry.initiatorType} ${resourceName}`,
+                op: `resource`,
+              });
+              resource.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+              resource.timestamp = resource.startTimestamp + duration;
+              // We remeber the entry script end time to calculate the difference to the first init mark
+              if (
+                entryScriptStartEndTime === undefined &&
+                entryScriptSrc &&
+                entryScriptSrc.indexOf(resourceName) > -1
+              ) {
+                entryScriptStartEndTime = resource.timestamp;
+              }
+              addSpan(resource);
+            }
+            break;
+          default:
+          // Yo
+        }
+      });
+
+    if (entryScriptStartEndTime !== undefined && tracingInitMarkStartTime !== undefined) {
+      const evaluation = transactionSpan.child({
+        description: 'evaluation',
+        op: `script`,
+      });
+      evaluation.startTimestamp = entryScriptStartEndTime;
+      evaluation.timestamp = tracingInitMarkStartTime;
+      addSpan(evaluation);
+    }
+    if (global.performance) {
```

Comment:
Here we check for `performance`, but above in [L428](https://github.com/getsentry/sentry-javascript/pull/2474/files#diff-a17621c348418aca8860e07b637d22a4R428) we don't.

Suggestion: check for `performance` at the top of `_addPerformanceEntries` and bail out early if it is not available.
In all other places no need to check for it again.

Reply:
Fixed